uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/calendar-ctrl.js

```diff
@@ -0,0 +1,283 @@
+define([
+  'leave-absences/shared/modules/controllers',
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/models/absence-period-model',
+  'leave-absences/shared/models/absence-type-model',
+  'leave-absences/shared/models/public-holiday-model',
+], function (controllers, _, moment) {
+  'use strict';
+
+  controllers.controller('CalendarCtrl', ['$timeout', 'shared-settings', 'AbsencePeriod', 'AbsenceType',
+    'PublicHoliday', 'OptionGroup', controller]);
+
+  function controller($timeout, sharedSettings, AbsencePeriod, AbsenceType, PublicHoliday, OptionGroup) {
+    this._dayTypes = [];
```

Comment:
I converted the private functions/variables to `_`. Previously private functions/variables were used if those are not required by the View, to make the scope lighter, now as we cant have private functions/variables with parent/child controller being introduced, I still kept the private functions/variables with `_` prefix to let the devs know that its now required by the View.